extension/fts/src/function/query_fts_index.cpp

```diff
@@ -21,22 +23,40 @@ using namespace kuzu::main;
 using namespace kuzu::function;
 
 struct QueryFTSBindData final : public FTSBindData {
-    std::string query;
+    std::shared_ptr<binder::Expression> query;
     const FTSIndexCatalogEntry& entry;
     QueryFTSConfig config;
 
     QueryFTSBindData(std::string tableName, common::table_id_t tableID, std::string indexName,
-        std::string query, const FTSIndexCatalogEntry& entry, std::vector<LogicalType> returnTypes,
-        std::vector<std::string> returnColumnNames, QueryFTSConfig config)
+        std::shared_ptr<binder::Expression> query, const FTSIndexCatalogEntry& entry,
+        std::vector<LogicalType> returnTypes, std::vector<std::string> returnColumnNames,
+        QueryFTSConfig config)
         : FTSBindData{std::move(tableName), tableID, std::move(indexName), std::move(returnTypes),
               std::move(returnColumnNames)},
           query{std::move(query)}, entry{entry}, config{std::move(config)} {}
 
+    std::string getQuery() const;
+
     std::unique_ptr<TableFuncBindData> copy() const override {
         return std::make_unique<QueryFTSBindData>(*this);
     }
 };
 
+std::string QueryFTSBindData::getQuery() const {
+    auto value = Value::createDefaultValue(query->dataType);
+    switch (query->expressionType) {
+    case ExpressionType::LITERAL: {
+        value = query->constCast<binder::LiteralExpression>().getValue();
+    } break;
+    case ExpressionType::PARAMETER: {
+        value = query->constCast<binder::ParameterExpression>().getValue();
+    } break;
```

Comment:
Isn't the switch case a bit weird, as both casted expressions call the same function? This looks like a polychromic use case to me.